app/react/Viewer/components/OCRStatus.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useEffect, useState } from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { FileType } from 'shared/types/fileType';
+import { Translate } from 'app/I18N';
+import { socket } from 'app/socket';
+import { bindActionCreators } from 'redux';
+import { postToOcr, getOcrStatus } from '../actions/ocrActions';
+import { ocrStatusTips } from '../utils/ocrStatusTips';
+import { reloadDocument } from '../actions/documentActions';
+
+type OCRStatusProps = {
+  file: FileType;
+};
+
+const formatDate = (time: number, locale: string) => {
+  const date = new Date(time);
+  return date.toLocaleString(locale);
+};
+
+const mapStateToProps = ({ settings, locale }: any) => {
+  const toggleOCRButton = settings.collection.get('ocrServiceEnabled');
+  return {
+    ocrIsToggled: toggleOCRButton || false,
+    locale,
+  };
+};
+
+const mapDispatchToProps = (dispatch: any) =>
+  bindActionCreators({ loadDocument: reloadDocument }, dispatch);
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type mappedProps = ConnectedProps<typeof connector>;
+type ComponentProps = OCRStatusProps & mappedProps;
+
+// eslint-disable-next-line max-statements
```

Comment:
This component can really be refactored.
Abstract more each case for the status, looks like most could be a simple function that returns a `div` with a `p` and a `Tranlsate`, only leave those that are really different (like the "noOcr" case) 
If there isn't enough time now, please create an issue to refactor this as tech debt so we don't leave it hanging there for too long.

Reply:
Will create the issue!